binary_search_tree/tree.py

```diff
@@ -14,35 +17,96 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(1)
+    # Space Complexity: O(N)
```

Comment:
Likewise, for a recursive implementation like this, the space used will be O(log(n)) because each recursion requires a bit of space on the stack as it recurses down.